data-protocols/dsp/dsp-catalog/dsp-catalog-http-dispatcher/src/main/java/org/eclipse/edc/protocol/dsp/catalog/dispatcher/delegate/CatalogRequestHttpDelegate.java

```diff
@@ -106,11 +110,18 @@ private String toJson(org.eclipse.edc.catalog.spi.protocol.CatalogRequestMessage
         try {
             var transformResult = transformerRegistry.transform(message, JsonObject.class);
             if (transformResult.succeeded()) {
-                return mapper.writeValueAsString(transformResult.getContent());
+                var compacted = compact(transformResult.getContent(), jsonLdContext());
+                return mapper.writeValueAsString(compacted);
             }
             throw new EdcException("Failed to write request.");
```

Comment:
I know this was not introduced in this PR but the exception should contain any error details from the transformResult. Otherwise, important context info may be lost. Can you file an issue so we can fix this in another PR?